### ISSUE TITLE
trilinos build error with xsdk+cuda

### ISSUE THREAD

**Issue by balay** [xsdk-project/xsdk-issues#232]
I don't remember if this is a new issue [I think this build succeeded in the prior weeks]

 xr_l_g113-xsdk-cuda-100-linux-gcc1131 

https://gitlab.com/xsdk-project/spack-xsdk/-/jobs/5276299076

[spack-build-out.txt](https://github.com/xsdk-project/xsdk-issues/files/12885216/spack-build-out.txt)

**Comment by jwillenbring** [comment]
@cgcgcg Any thoughts on this error, or a suggestion for who else might best weigh in on this? I am seeing a lot of issues like:

from /data/xsdkci/VS1mJQ1K/1/xsdk-project/spack-xsdk/spack-stage/balay/spack-stage-trilinos-14.4.0-3uzqb2ecpsyoggaotse64jvx6b6c22cb/spack-src/packages/amesos2/src/Amesos2_Factory.cpp:44:
/apps/spacks/2023-05-19/opt/spack/linux-rocky9-x86_64/gcc-9.5.0/cuda-11.7.1-lwifpdazrh7hzn5b73qere6buklzbefv/include/cuda_bf16.hpp:547:37: error: conflicting declaration of C function 'float SLUD::__low2float(SLUD::__nv_bfloat162)'
  547 | __CUDA_HOSTDEVICE_BF16_DECL__ float __low2float(const __nv_bfloat162 a)
      |                                     ^~~~~~~~~~~

**Comment by cgcgcg** [comment]
@srajama1 @lucbv The errors in the build log look like they are coming out of Kokkos Kernels.

**Comment by lucbv** [comment]
Yeah, we can have a look, conflicting declaration error should not be to hard to handle.

**Comment by pghysels** [comment]
Can this be fixed in the Amasos2 package?
Do the superlu_dist headers need to be included from within the `SLUD` namespace?:
https://github.com/trilinos/Trilinos/blob/d1f042da58a3e1de8bb198a5bfc3821608db625b/packages/amesos2/src/Amesos2_Superludist_TypeMap.hpp#L79

**Comment by jwillenbring** [comment]
@lucbv Any update on this one?

**Comment by cgcgcg** [comment]
Is this the same as #235 ? If so, this is being worked on.

**Comment by balay** [comment]
I'll just note that we are close to the release of xsdk-1.0.0, so any fixes [or changes] would be for the next release cycle.

even if trilinos issues are fixed - we would need to go through the process of  (a) getting the fixes to spack (b) re-testing and looking for other dependent package breakages (with `trilinos+cuda` or `trilinos+rocm`) - that would again need another cycle of fixes and testing. 
 
Right now we have `trilinos~cuda~rocm` in `xsdk-1.0.0` release [i.e the same as prior xsdk release cycles]
